docs/typescript-styleguide.md

```diff
@@ -52,19 +52,28 @@ export interface MyComponentProps {
   exampleOptionalProp?: string;
 }
 
-declare const MyComponent: React.FunctionComponent<MyComponentProps>;
+declare function MyComponent(props: MyComponentProps): JSX.Element;
 
 export default MyComponent;
 ```
 
+Component could also be declared as `FunctionComponent` when it uses the `children` prop:
```

Comment:
yet it's a valid type for most of the cases, it might be problematic if you use that in every function component as it adds an optional children prop, but that type will accept no children by default  in version `@types/react@^18.0.0`.

Reply:
on second thought I'll just remove that